Carbonate/PushReactable.cs

```diff
@@ -0,0 +1,144 @@
+// <copyright file="PushReactable.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Carbonate;
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using Core;
+using Services;
+
+/// <summary>
+/// Defines a provider for push-based notifications.
+/// </summary>
+public sealed class PushReactable : ReactableBase<IReceiveReactor>, IPushable
+{
+    private readonly ISerializerService serializerService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PushReactable"/> class.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    [SuppressMessage("ReSharper", "UnusedMember.Global", Justification = "Public API for users.")]
+    public PushReactable() => this.serializerService = new JsonSerializerService();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PushReactable"/> class.
+    /// </summary>
+    /// <param name="serializerService">The serializer used to serialize the messages.</param>
+    public PushReactable(ISerializerService serializerService) =>
+        this.serializerService = serializerService ??
+            throw new ArgumentNullException(nameof(serializerService), "The parameter must not be null.");
+
+    /// <inheritdoc/>
+    public void Push(Guid eventId) => SendNotifications(null, eventId);
+
+    /// <inheritdoc/>
+    /// <exception cref="ObjectDisposedException">Thrown if this method is invoked after disposal.</exception>
+    public void PushData<T>(in T data, Guid eventId)
+    {
+        if (IsDisposed)
+        {
+            throw new ObjectDisposedException(nameof(PushReactable), $"{nameof(PushReactable)} disposed.");
+        }
+
+        try
+        {
+            var jsonData = this.serializerService.Serialize(data);
+
+            var message = new JsonMessage(this.serializerService, jsonData);
+
+            SendNotifications(message, eventId);
+        }
+        catch (JsonException e)
+        {
+            SendError(e, eventId);
+        }
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="ObjectDisposedException">Thrown if this method is invoked after disposal.</exception>
+    public void PushMessage(in IMessage message, Guid eventId) => SendNotifications(message, eventId);
+
+    /// <summary>
+    /// Sends a notification to all of the <see cref="IReactor"/>s that match the given <paramref name="eventId"/>.
```

Comment:
Should the "s" be after the > after IReactor?
```suggestion
    /// Sends a notification to all of the <see cref="IReactor"/>s that match the given <paramref name="eventId"/>.
```

Reply:
Yes.  I did this to have it naturally make sense when reading it.